system/system_test.go

```diff
@@ -182,6 +179,97 @@ func TestPing(t *testing.T) {
 	}
 }
 
+func TestReboot(t *testing.T) {
+	tests := []struct {
+		desc               string
+		op                 *system.RebootOperation
+		want               *spb.RebootResponse
+		rebootErr, wantErr string
+		statusErr          []error
+		cancelContext      bool
+	}{
+		{
+			desc: "Test reboot",
+			op: system.NewRebootOperation().RebootMethod(spb.RebootMethod_COLD).Subcomponents([]*tpb.Path{{
+				Elem: []*tpb.PathElem{
+					{Name: "components"},
+					{Name: "component", Key: map[string]string{"name": "RP0"}},
+				},
+			}}),
+			want: &spb.RebootResponse{},
+		},
+		{
+			desc: "Test reboot wait for active status",
+			op:   system.NewRebootOperation().RebootMethod(spb.RebootMethod_COLD).WaitForActive(true),
+			want: &spb.RebootResponse{},
+		},
+		{
+			desc:      "Test reboot wait for active status and ignore unavailable error",
+			op:        system.NewRebootOperation().RebootMethod(spb.RebootMethod_COLD).WaitForActive(true).IgnoreUnavailableErr(true),
+			statusErr: []error{status.Errorf(codes.Unavailable, "unavailable")},
+			want:      &spb.RebootResponse{},
+		},
+		{
+			desc:      "Test reboot wait for active status returns unknown error",
+			op:        system.NewRebootOperation().RebootMethod(spb.RebootMethod_COLD).WaitForActive(true).IgnoreUnavailableErr(true),
+			statusErr: []error{status.Errorf(codes.Unknown, "unknown")},
+			wantErr:   "unknown",
+		},
+		{
+			desc:      "Test reboot returns error on reboot",
+			op:        system.NewRebootOperation().RebootMethod(spb.RebootMethod_COLD),
+			rebootErr: "Reboot operation error",
+			wantErr:   "Reboot operation error",
+		},
+		{
+			desc:      "Test reboot returns error on reboot status",
+			op:        system.NewRebootOperation().RebootMethod(spb.RebootMethod_COLD).WaitForActive(true),
+			statusErr: []error{status.Errorf(codes.Unavailable, "unavailable")},
+			wantErr:   "unavailable",
+		},
+		{
+			desc:          "Test reboot with context cancel",
+			op:            system.NewRebootOperation().RebootMethod(spb.RebootMethod_COLD).WaitForActive(true),
+			wantErr:       "context",
+			cancelContext: true,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.desc, func(t *testing.T) {
+			var fakeClient internal.Clients
+			fakeClient.SystemClient = &fakeSystemClient{
+				RebootFn: func(context.Context, *spb.RebootRequest, ...grpc.CallOption) (*spb.RebootResponse, error) {
+					if tt.rebootErr != "" {
+						return nil, fmt.Errorf(tt.rebootErr)
+					}
+					return tt.want, nil
+				},
+				RebootStatusFn: func(context.Context, *spb.RebootStatusRequest, ...grpc.CallOption) (*spb.RebootStatusResponse, error) {
+					if len(tt.statusErr) > 0 {
+						statusErr := tt.statusErr[0]
+						tt.statusErr = tt.statusErr[1:]
+						return nil, statusErr
+					}
+					return &spb.RebootStatusResponse{Active: true}, nil
```

Comment:
We need a case where we get a non-nil rebootstatusresponse that is not yet active before we get an active response. I think you'll need to keep a statusResps slice in the fixture similar to your statusErrs field

Reply:
done